src/main/java/com/google/cloudassets/discovery/assetobjects/AppAppEngineObject.java

```diff
@@ -49,18 +49,23 @@ public Builder(Map<String,Object> assetProperties, ProjectConfig projectConfig)
         public AppAppEngineObject build() {
             // Set AssetObject fields
             setKind(AssetKind.APP_APP_ENGINE_ASSET);
-            setName(assetProperties.get("name"));
-            setId(assetProperties.get("id"));
-            setLocation(getLastSeg(assetProperties.get("locationId")));
-            setStatus(assetProperties.get("servingStatus"));
+            try {
+                setName(assetProperties.get("name"));
+                setId(assetProperties.get("id"));
+                setLocation(getLastSeg(assetProperties.get("locationId")));
+                setStatus(assetProperties.get("servingStatus"));
 
-            // Set specific asset type fields
-            specificObjectClass.authDomain = castToString(assetProperties.get("authDomain"));
-            specificObjectClass.defaultHostname = castToString(assetProperties.get("defaultHostname"));
-            specificObjectClass.codeBucket = castToString(assetProperties.get("codeBucket"));
-            specificObjectClass.gcrDomain = castToString(assetProperties.get("gcrDomain"));
-            specificObjectClass.defaultBucket = castToString(assetProperties.get("defaultBucket"));
-            specificObjectClass.databaseType = castToString(assetProperties.get("databaseType"));
+                // Set specific asset type fields
+                specificObjectClass.authDomain = castToString(assetProperties.get("authDomain"));
+                specificObjectClass.defaultHostname = castToString(assetProperties.get("defaultHostname"));
+                specificObjectClass.codeBucket = castToString(assetProperties.get("codeBucket"));
+                specificObjectClass.gcrDomain = castToString(assetProperties.get("gcrDomain"));
+                specificObjectClass.defaultBucket = castToString(assetProperties.get("defaultBucket"));
+                specificObjectClass.databaseType = castToString(assetProperties.get("databaseType"));
+            } catch (NullPointerException exception) {
+                logger.atInfo().withCause(exception).log("Could not set all of the AppAppEngineObject " +
```

Comment:
If you define your own "get" method, the exception message will be able to include the exception include the name of the missing property and also make this method shorter and more readable.

Reply:
That is a good idea but the problem is that the get method does not throw the NullPointerException it just returns null and then the function that uses this value throws the NullPointerException.
I can add a try & catch statement to helper functions defined in the AssetObject class if you think it makes more sense (but then we still do not have the name of the missing property). 
WDYT?